src/ws/main.c

```diff
@@ -146,11 +147,10 @@ main (int argc,
 
   context = g_option_context_new (NULL);
   g_option_context_add_main_entries (context, cmd_entries, NULL);
-
-  if (!g_option_context_parse (context, &argc, &argv, error))
-    {
-      goto out;
-    }
+  success = g_option_context_parse (context, &argc, &argv, error);
+  g_option_context_free (context);
```

Comment:
Why not just include this under the `out:` block with everything else?

....also, remind me again why we're not using g_autoptr()?

Reply:
I had it in `out:` at first, but there's no need to keep the context allocated throughout ws'es lifetime.

We can use g_autoptr now, but that would be functionally equivalent of free'ing it in `out:`, right?

Reply:
This seems like a pretty small optimisation to sacrifice idiomatic readability for....

Reply:
Ack. I moved to g_auto now, and for consistency did that for the other applicable variables as well.